### DISCUSSION TITLE
Issue in loading the dbever project into eclipse.

### DISCUSSION THREAD

**Discussion by kolitha1981** [dbeaver/dbeaver#18797] [Q&A]
Hi, I have cloned the repo and have built the project using maven when I import the project into eclipse Ide the projects their build paths of the projects are not getting set properly as shown below.

![image](https://user-images.githubusercontent.com/5469747/212924908-64f0e73f-89db-40e1-aa5b-73a33291b0fe.png)

Any help you can provide on this?

**Comment by E1izabeth** [comment]
We don't use this option to build a separate plugins. There is a wiki about setting up your development environment https://github.com/dbeaver/dbeaver/wiki/Develop-in-Eclipse . Hope it will help

**Reply by gitxpy** [reply]
following up on this wiki guide since there seems to have a bunch of problems with eclipse...

On eclipse 2022-12  ( linux RH7, openjdk version "17" 2021-09-14,  maven 3.8.7)
- I get 5587 errors, mostly of  "unresolved type" from 20+ different projects

- using "maven" -> "update project"   does not fix the problems

any idea ?
is it platform dependant ?
could you specify on which you've actually managed to build properly ?


ps.
 could someone also update this page ?
  https://github.com/dbeaver/dbeaver/wiki/Build-from-sources

**Reply by chirontt** [reply]
@gitxpy You should use the [Eclipse IDE for RCP and RAP Developers](https://www.eclipse.org/downloads/packages/release/2022-12/r/eclipse-ide-rcp-and-rap-developers), not the standard Eclipse IDE for Java, to avoid the compile errors you mentioned. The wiki guide does mention this requirement:

> You will need to setup [Eclipse IDE for RCP and RAP Developers](https://www.eclipse.org/downloads/packages/release/2022-12/r/eclipse-ide-rcp-and-rap-developers). The current default Eclipse version is _2022-12...

**Reply by gitxpy** [reply]
thanx for the answer.

- I'm developping a RCP app, so I think I had the RCP/RAP package ;)
- for the sake of certitude, I reinstalled it from .tar.gz and reapplied the wiki guide  (only noticeable diffrence being in step 10, it only show maven-clean-plugin)

=> result is the same (~5000 errors)

-- trying to dig in, the first project in error in the hierarchical list is  plugins > org.jkiss.dbeaver.data.gis
--- complaining in MANIFEST.MF  not being able to resolved the  Require-bundle: org.jkiss.bundle.gis
--- (indeed, can't see it in   dbeaver>bundles>... )

any idea how to resolve this ? (similar as issue #5991 ? )

thanx

**Reply by chirontt** [reply]
Missing the `org.jkiss.bundle.gis` bundle? It's an `external` dependency which is available from `http://dbeaver.io/eclipse-repo/` site.

I suspect you haven't done the steps 1 - 5 of the [Develop-in-Eclipse](https://github.com/dbeaver/dbeaver/wiki/Develop-in-Eclipse) wiki article, to set up your workspace properly for DBeaver development.

**Reply by gitxpy** [reply]
thanx for following up.

I did follow the steps, after this first warning
![image](https://user-images.githubusercontent.com/18109963/217177331-db755402-f167-4df1-b9eb-16ed88c64f8b.png)
it seems to go through and ask to restart eclipse ... 
(but still this problem of unresolved bundle)

any idea how to narrow this down and solve it ?

**Reply by chirontt** [reply]
Somehow your Eclipse installation has some clashes with the external dependencies provided by the `http://dbeaver.io/eclipse-repo/` site when you try to add them to your workspace, resulting in many of those dependencies not being installed, hence 'missing' and thus causing compile errors in the `dbeaver` project in the workspace. I don't get this problem at all with my workspace, so I can't help you there. But for your information, here are the simple steps I've done to set it up and they work for me:

- download the Eclipse RCP version 2012-12, from the eclipse.org site, and extract it to an empty folder
- create a new folder to use as a workspace for the DBeaver project
- start up the Eclipse RCP and select the above empty folder as workspace
- then follow the instructions (all the steps) in the [Develop in Eclipse](https://github.com/dbeaver/dbeaver/wiki/Develop-in-Eclipse) wiki guide. There would be no errors/problems in following the steps.

I've done the above steps on at least 3 machines (x64 Ubuntu Linux, arm64 Debian Linux, x64 and arm64 Windows) and they all work fine for me.

**Reply by gitxpy** [reply]
I guess we can focus on the dbeaver repo problem ...
and thanks for the tip, indeed **on a fresh new workspace**; **it works just fine** !
so, problem solved :)

(slightly off-topic, the instructions for Running / Debugging seem outdated / misleading)

And here are some clues for people who might encountered the symptoms, from a fresh install, I first got  :

Cannot complete the install because one or more required items could not be found.
  Software being installed: SSHv2 library for Java 0.34.1 (org.jkiss.bundle.sshj 0.34.1)
  Missing requirement: SSHv2 library for Java 0.34.1 (org.jkiss.bundle.sshj 0.34.1) requires 'java.package; org.bouncycastle.openssl 0.0.0' but it could not be found

![image](https://user-images.githubusercontent.com/18109963/217479689-092b2228-5824-4029-ae7e-1c7e9a38f4f9.png)
luckily, on the second go, it went through (!?)


On my original workspace, still have build errors, and on install new software,  I get :

Cannot complete the request.  See the error log for details.
"Mockito Mock Library for Java. Core bundle requires Byte Buddy and Objenesis." will be ignored because a newer version is already installed. 
"Byte Buddy (without dependencies)" will be ignored because a newer version is already installed. 
"Apache Ant minimum package" will be ignored because a newer version is already installed. 
"Byte Buddy agent" will be ignored because a newer version is already installed. 
"Apache Commons Logging" will be ignored because a newer version is already installed. 
"Apache Commons CLI" will be ignored because a newer version is already installed. 
"Objenesis" will be ignored because a newer version is already installed.
